### PR TITLE
Convert to Kotlin

### PR DESCRIPTION
**Pull request by brendan-fahy**
* Lower minSdk version

* God Damn onResume

* Better handling (presumably) of the Quick Settings tile

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,10 +1,12 @@
 apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
 
 android {
     compileSdkVersion 26
     defaultConfig {
         applicationId "com.mbcdev.goddamn"
-        minSdkVersion 24
+        minSdkVersion 23
```
Comment (baz8080) [inline app/build.gradle]: I think 24 is the minimum
https://developer.android.com/reference/android/service/quicksettings/Tile.html

File: app/src/main/kotlin/goddamn/GodDamnService.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Just stop judging me. It's your fault Egan and Brenda.
+ *
+ * It wasn't my fault before, but some of it is now.
+ *
+ * Created by bcarroll on 13/11/2017.
+ */
+@RequiresApi(api = Build.VERSION_CODES.N)
+class GodDamnService : TileService() {
```
Comment (baz8080) [inline app/src/main/kotlin/goddamn/GodDamnService.kt:16]: Would like to test this :) 
there are some lifecycle methods in TileService that might help too

**Review by baz8080 — CHANGES_REQUESTED** [review]
ohyou.gif

File: app/build.gradle (hunk 1 above)
Comment (brendan-fahy) [reply app/build.gradle]: FINE

File: app/src/main/kotlin/goddamn/GodDamnService.kt (hunk 2 above)
Comment (brendan-fahy) [reply app/src/main/kotlin/goddamn/GodDamnService.kt:16]: Yeah it was mostly me being too lazy to plug out my dev phone and plug in my real phone. Will try it now.

**Review by baz8080 — APPROVED** [review]

**Pull request merged**